common/Semicolons.yml

```
@@ -1,7 +1,7 @@
 extends: existence
-message: "Use semicolons judiciously."
+message: "Use semicolons prudently."
```

```suggestion
message: "Avoid (over)use of semicolons`
```

Or something like that

---

``Avoid (over)use`` doesn't seem to work since they aren't supposed to use it at all except for complicated lists.

How about ``Only use semicolons in complicated lists with punctuation.`` so that we also account for the issue I detailed above.

---

Hey, as usual when facing a vale issue based on grammar/style I tend to check what is available in official vale package

we can find these:
https://github.com/errata-ai/Google/blob/28adc48796277429c7d0cc80932569c5e7b0c360/Google/Semicolons.yml#L4

This one is clearly the source of the existing file

And this one, I like the Microsoft one
https://github.com/errata-ai/Microsoft/blob/c798f131106d77c136a3c1b5fa4916dee1de8edd/Microsoft/Semicolon.yml#L4

I was surprised to find nothing in write-good and prose-lint

https://github.com/errata-ai/proselint
https://github.com/errata-ai/write-good

---

@alexvonme Im unsure if you saw my message about SemiColomn in Microsoft package

their message is very general

```yaml
message: "Try to simplify this sentence."
```

---

@ccoVeille Yes, Microsoft has a different messaging because they distinctly [allow semicolons in multiple use cases](https://learn.microsoft.com/en-us/style-guide/punctuation/semicolons), whereas SUSE explicitly [only allows it in "in place of commas in very complicated series"](https://documentation.suse.com/style/current/single-html/docu_styleguide/#sec-semicolon). As such, I don't think we should adopt a similarly general message.

---

ok, I get it now